components/productList/FilteredProductList/sorting.ts

```diff
@@ -0,0 +1,57 @@
+import { OrderDirection, ProductOrderField } from "@/saleor/api";
+
+export interface UrlSorting {
+  field: ProductOrderField;
+  direction: OrderDirection;
+}
+
+export interface SortingOption {
+  label: string;
+  field?: ProductOrderField;
+  direction?: OrderDirection;
+  chosen: boolean;
+}
+
+export const getSortingOptions = (chosen: UrlSorting | null) => {
```

Comment:
Use map instead of lets + for. Also, since `chosen` is later on a boolean prop in the options object, the name `chosen` for what is in fact a urlSorting object is misleading. Let's call it `urlSorting` maybe?